File: 3.swift
func priority(_ item: Character) -> Int {
    let chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
    return chars.firstIndex(of: item)?.utf16Offset(in: chars) ?? 0 + 1 
}
var rucksacks: [String] = []
var p1 = 0, p2 = 0
while let rucksack = readLine() {
    let first = rucksack.prefix(rucksack.count / 2), second = rucksack.suffix(rucksack.count / 2)
    for item in  first {
        if second.contains(item) {
            p1 += priority(item)
            break
        }
    }
    rucksacks.append(rucksack)
    if rucksacks.count == 3 {
        for item in rucksacks[0] {
            if rucksacks[1].contains(item) && rucksacks[2].contains(item) {
                p2 += priority(item)
                break
            }
        }
        rucksacks = []
    }
}
print(p1, p2)

File: 3.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func priority(c rune) int {
	if c >= 'a' && c <= 'z' {
		return int(c-'a') + 1
	}
	if c >= 'A' && c <= 'Z' {
		return int(c-'A') + 27
	}
	return 0
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	p1, p2 := 0, 0
	items := make([]string, 0)
	for scanner.Scan() {
		item := scanner.Text()
		first, second := item[:len(item)/2], item[len(item)/2:]
		for _, c := range first {
			if strings.Contains(second, string(c)) {
				p1 += priority(c)
				break
			}
		}
		items = append(items, item)
		if len(items) == 3 {
			for _, r := range items[0] {
				if strings.Contains(items[1], string(r)) && strings.Contains(items[2], string(r)) {
					p2 += priority(r)
					break
				}
			}
			items = make([]string, 0)
		}
	}
	fmt.Println(p1, p2)
}
